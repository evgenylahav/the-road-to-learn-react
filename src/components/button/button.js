import * as React from 'react';
import PropTypes from 'prop-types';

export const Button = ({ onClick, className, children }) => {
    return(
      <button
        onClick={onClick}
        className={className}
        type="button"
      >
        {children}
      </button>
    );
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};