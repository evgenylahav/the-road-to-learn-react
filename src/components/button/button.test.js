import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Button } from './button';
import { shallow } from 'enzyme';
import { testNameToKey } from 'jest-snapshot/build/utils';


describe('Button tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Click me</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('shows button children', () => {
    const element = shallow(
      <Button className={'test-button'}>Click me</Button>
    );
    expect(element.find('.test-button').length).toBe(1);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Button>Click me</Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
