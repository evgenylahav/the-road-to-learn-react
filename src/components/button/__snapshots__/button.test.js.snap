// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests has a valid snapshot 1`] = `
<button
  type="button"
>
  Click me
</button>
`;
